pkg/agent/billing/clients.go

```diff
@@ -84,7 +85,7 @@ func createClients(ctx context.Context, logger *zap.Logger, cfg ClientsConfig) (
 			Name:            "s3",
 			Base:            client,
 			BaseConfig:      c.BaseClientConfig,
-			NewBatchBuilder: jsonArrayBatch(reporting.NewGZIPBuffer),
+			NewBatchBuilder: jsonLinesBatch(reporting.NewGZIPBuffer),
```

Comment:
I think, we already once had incident when the format was not exactly like we expected it to be.

May i ask you to add a simple unit/integration test, to double validate that produced data format matches the expectations?

Reply:
I've fixed e2e tests and added unit tests.